pkg/ingester/ingester.go

```diff
@@ -2151,6 +2151,13 @@ func (i *Ingester) createTSDB(userID string, walReplayConcurrency int) (*userTSD
 	}
 	db.DisableCompactions() // we will compact on our own schedule
 
+	// Register metric that reports Head MaxTime in seconds.
+	// (TSDB library already exports prometheus_tsdb_head_max_time metric, but it's unit-neutral.)
+	promauto.With(tsdbPromReg).NewGaugeFunc(prometheus.GaugeOpts{
```

Comment:
Now it looks quite a complicated wiring. Have you considered just defining the GaugeFunc in the main registry (not the per-TSDB one), iterate through TSDBs and get the max head time value?

Reply:
I don't see this very complicated, but your suggestion is certainly a possibility as well, and will be simpler. Let me do that.

Reply:
I've implemented this change in https://github.com/grafana/mimir/pull/5815/commits/3b8c2a79c72f35b04aef73a75f06b4c4a31e1728.